starknet_py/contract_abi_resolver.py

```diff
@@ -0,0 +1,131 @@
+# Needed because of string typed Contract
+# pyright: reportUndefinedVariable=false
+import warnings
+from enum import Enum
+from typing import Union, List, TypedDict, Tuple
+
+from starknet_py.net.client import Client
+from starknet_py.net.client_errors import ContractNotFoundError, ClientError
+from starknet_py.net.client_models import Abi, DeclaredContract
+from starknet_py.net.models import Address
+from starknet_py.proxy_check import (
+    ProxyResolutionError,
+    OpenZeppelinProxyCheck,
+    ArgentProxyCheck,
+    ProxyCheck,
+)
+
+
+class ProxyConfig(TypedDict, total=False):
+    """
+    Proxy resolving configuration
+    """
+
+    proxy_checks: List[ProxyCheck]
+    """
+    List of classes implementing :class:`starknet_py.proxy_check.ProxyCheck` ABC,
+    that will be used for checking if contract at the address is a proxy contract.
+    """
+
+
+def prepare_proxy_config(proxy_config: Union[bool, ProxyConfig]) -> ProxyConfig:
+    if isinstance(proxy_config, bool):
+        if not proxy_config:
+            return ProxyConfig()
+        proxy_config = ProxyConfig()
+
+    if "max_steps" in proxy_config:
+        warnings.warn(
+            "ProxyConfig.max_steps is deprecated. Contract.from_address always makes at most 1 step.",
+            category=DeprecationWarning,
+        )
+    proxy_checks = [OpenZeppelinProxyCheck(), ArgentProxyCheck()]
+    return {"proxy_checks": proxy_checks, **proxy_config}
+
+
+class ImplementationType(Enum):
+    """
+    Enum representing transaction types
+    """
+
+    CLASS_HASH = "class_hash"
+    ADDRESS = "address"
+
+
+class ContractAbiResolver:
```

Comment:
Instead of passing proxy config to multiple methods you can just make them regular object methods and accept parameter. This is a more pythonic way.